src/dlt_typed_payload.rs

```diff
@@ -13,5 +13,6 @@ pub enum DltTypedPayload<'a> {
         info: Option<DltMessageInfo>,
         msg_id: u32,
         payload: &'a [u8],
+        control_message: Option<ControlMessage<'a>>,
```

Comment:
On second look at `DltTypedPayload`, the payload should probably be redesigned to allowed so the user can easily filter based on the type of message received.

So instead of adding a new field I think we should extend the values of `DltTypedPayload` and split them up based on the message type. E.g.

```
pub enum DltTypedPayload<'a> {
     /// Generic non verbose message without message type infos.
     GenericNv(...),
     /// Verbose log message.
     LogV(...),
     /// Non verbose log message.
     LogNv(...),
     /// Verbose trace message.
     TraceV(...),     
     /// Non verbose trace message.
     TraceNv(...),
     /// Verbose network message.
     NetworkV(...),
     /// Non verbose network message.
     NetworkNv(...),
     /// Verbose control message.
     ControlV(...),
     /// Non verbose control message.
     ControlNv(...),
}
```

Also we should probably create structs for the different value types, meaning replacing `LogNv { info: Option<DltMessageInfo>, ... }` with `LogNv(LogNvPayload<'a>)`. That way explicitly skipping values (e.g. `ControlNv(_) => {}`) is not as verbose as it is now the baked in structs fields in the enum.